gimie/sources/github.py

```diff
@@ -80,7 +100,9 @@ def _request(cls, query_path: str) -> Any:
             The query, without the base path.
 
         """
-        resp = requests.get(f"{GH_API}/{query_path.lstrip('/')}")
+        resp = requests.get(
+            f"{GH_API}/{query_path.lstrip('/')}", headers=GH_HEADERS
```

Comment:
Maybe `_request` could take an optional `headers` option and call `_set_auth` only if it is not provided?
This would allow user to provide tokens through the API when it is not convenient to use `.env` or env variables.

Then we could provide some way for the user to provide a token or headers, through one of the following (not sure which is best):
1. `GithubExtractor('https://...', token="....")`
2. `GithubExtractor('https://...', headers={"...": "..."})`
3. `GithubExtractor('https://...').extract(token="...")`
4. `GithubExtractor('https://...').extract(headers={"...": "..."})`

Reply:
I opted for this approach: `GithubExtractor('https://...', github_token="....")`

Reply:
The token as attribute makes sense to me as well. One thing I am not sure about is making `_set_auth` `@classmethod`:

Let's imagine I instantiate 15 `GithubExtractor`'s in my program (e.g. crawling different organizations in parallel). In the current implementation, `github_token` is a class attribute, doesn't it mean its state is shared across all instances ?

If so, I would not be able to set different tokens for different instances.on the other hand,  there may be scenarios where it is better to have the token updated in all instances... 

Depending on the preferred use case, we could bind the token to the class or the instance. I am not sure which is best, what do you think @sabinem @martinfontanet ?